### ISSUE TITLE
Final fields are excluded from the API jar

### ISSUE THREAD

**Issue by astellingwerf** [gradle/gradle#819]
When a JVM module has no explicit API, package private members are included in the API jar. However, those members are included IAOI it either has:
- Zero flags
- Is only static
- Is only from a super class
- Is only static from a super class

This means that for example `final` fields are not included.

### Expected Behavior
Like for `public` and `protected`, I'd expect all package private fields are included in the API, regardless of whether or not they are `final`, `volatile`, `native`, `synchronized`, or whatever.

### Current Behavior
For `public` and `protected`, a bit flag check is performed, ignoring all other flags, whereas for package private the set is rather limited (see `org.gradle.jvm.tasks.api.internal.ApiMemberSelector`):
```
    private static boolean isPublicMember(int access) {
        return (access & ACC_PUBLIC) == ACC_PUBLIC;
    }

    private static boolean isProtectedMember(int access) {
        return (access & ACC_PROTECTED) == ACC_PROTECTED;
    }

    private static boolean isPackagePrivateMember(int access) {
        return access == 0
            || access == ACC_STATIC
            || access == ACC_SUPER
            || access == (ACC_SUPER | ACC_STATIC);
    }
```

### Steps to Reproduce
1. Setup
```
git clone https://github.com/astellingwerf/package-private-final-gradle-issue.git
cd package-private-final-gradle-issue
```
2. Prove that it works for non-final fields:
`gradle clean build`
3. Reproduce the bug. Make the field in `src/main/java/nl/astellin/Main.java` `final` and rebuild
`gradle clean build`

**Comment by astellingwerf** [comment]
I didn't test this, but I guess the fix would be like this:
```
    private static boolean isPackagePrivateMember(int access) {
        return access & (ACC_PUBLIC | ACC_PROTECTED | ACC_PRIVATE) == 0;
    }
```

**Comment by melix** [comment]
@astellingwerf It looks like a reasonable fix, yes. The case where we declare no API is a bit messy, and we have to do this for backwards compatibility, and you have a good point here, thanks!

**Comment by ThiporKong** [comment]
Because #925 and #928 have been closed for being redundant to this issue, I'd be grateful if a fix for #819 will also be checked against the tests I have provided (I'm kind of sceptic, because I did describe only symptoms, while #819 jumps straight into conclusions about the implementation):

From #925:
```gradle
plugins {
    id 'jvm-component'
    id 'java-lang'
}

model {
    components {
        def aSrc = Files.createTempDirectory("a-src").toFile()
        new File(aSrc, "A.java").text = """
public class A {
    public static final String HALLO = "Hallo";
    public static final int HELLO = 4711;
}
"""

        def bSrc = Files.createTempDirectory("b-src").toFile()
        new File(bSrc, "B.java").text = """
public class B {
    public static void main(String[] args) {
        switch (args[0]) {
        case A.HALLO:
        }
        
        switch (args.length) {
        case A.HELLO:
        }
    }
}
"""

        a(JvmLibrarySpec) {
            sources.java.source.srcDir aSrc
        }
        b(JvmLibrarySpec) {
            sources.java.source.srcDir bSrc
            dependencies.library 'a'
        }
    }
}
```

and from #928:

```gradle
plugins {
    id 'jvm-component'
    id 'java-lang'
}

model {
    components {
        def aSrc = Files.createTempDirectory("a-src").toFile()
        new File(aSrc, 'A1.java').text = 'abstract class A1 {}'
        new File(aSrc, 'A2.java').text = 'public class A2 extends A1 {}'

        def bSrc = Files.createTempDirectory("b-src").toFile()
        new File(bSrc, 'B.java').text = 'public class B extends A2 {}'

        a(JvmLibrarySpec) {
            sources.java.source.srcDir aSrc
        }
        b(JvmLibrarySpec) {
            sources.java.source.srcDir bSrc
            dependencies.library 'a'
        }
    }
}
```
